### DISCUSSION TITLE
Ensure that upstream assets are materialized

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17872] [Q&A] [answered]
I’m explicitly defining asset dependencies like so:

`ins={'code_of_conduct_extracted': AssetIn(key=['llm', 'extracted', 'extracted_code_of_conduct'])},`

How do I ensure all upstream assets are also being materialized whether in Dagit or via python?

Right now, I get this warning: “Materializing these assets may fail because the upstream assets listed below have not been materialized yet.”

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699551484705199?thread_ts=1699551484.705199&cid=C01U954MEER).*

**Comment by jamiedemaria** [comment, marked as answer]
It depends on how you are running the asset. If you are in the global asset graph, clicking an asset, then clicking the "Materialize" button, there isn't a way to automatically run the parent assets as well. You'd need to also select those assets before clicking "Materialize"

If you are executing the asset in a job or from a schedule or sensor, you can use the `AssetSelection` API ([docs](https://docs.dagster.io/_apidocs/assets#dagster.AssetSelection)) to also include the parents in the selection

**Reply by seanlindo** [reply]
ty
